editor/editor_build_profile.cpp

```diff
@@ -79,38 +104,215 @@ const bool EditorBuildProfile::build_option_disabled_by_default[BUILD_OPTION_MAX
+
+/* clang-format off */
+
+const HashMap<EditorBuildProfile::BuildOption, LocalVector<EditorBuildProfile::BuildOption>> EditorBuildProfile::build_option_dependencies = {
+	{ BUILD_OPTION_OPENXR, {
+			BUILD_OPTION_XR,
+	} },
+	{ BUILD_OPTION_FORWARD_RENDERER, {
+			BUILD_OPTION_RENDERING_DEVICE,
+	} },
+	{ BUILD_OPTION_MOBILE_RENDERER, {
+			BUILD_OPTION_RENDERING_DEVICE,
+	} },
+	{ BUILD_OPTION_VULKAN, {
+			BUILD_OPTION_FORWARD_RENDERER,
+			BUILD_OPTION_MOBILE_RENDERER,
+	} },
+	{ BUILD_OPTION_D3D12, {
+			BUILD_OPTION_FORWARD_RENDERER,
+			BUILD_OPTION_MOBILE_RENDERER,
+	} },
+	{ BUILD_OPTION_METAL, {
+			BUILD_OPTION_FORWARD_RENDERER,
+			BUILD_OPTION_MOBILE_RENDERER,
+	} },
+	{ BUILD_OPTION_PHYSICS_GODOT_2D, {
+			BUILD_OPTION_PHYSICS_2D,
+	} },
+	{ BUILD_OPTION_PHYSICS_GODOT_3D, {
+			BUILD_OPTION_PHYSICS_3D,
+	} },
+	{ BUILD_OPTION_PHYSICS_JOLT, {
+			BUILD_OPTION_PHYSICS_3D,
+	} },
+	{ BUILD_OPTION_DYNAMIC_FONTS, {
+			BUILD_OPTION_TEXT_SERVER_ADVANCED,
+	} },
+	{ BUILD_OPTION_WOFF2_FONTS, {
+			BUILD_OPTION_TEXT_SERVER_ADVANCED,
+	} },
+	{ BUILD_OPTION_GRAPHITE_FONTS, {
+			BUILD_OPTION_TEXT_SERVER_ADVANCED,
+	} },
+};
+
+const HashMap<EditorBuildProfile::BuildOption, LocalVector<String>> EditorBuildProfile::build_option_classes = {
+	{ BUILD_OPTION_3D, {
+			"Node3D",
+	} },
+	{ BUILD_OPTION_NAVIGATION_2D, {
+			"NavigationAgent2D",
+			"NavigationLink2D",
+			"NavigationMeshSourceGeometryData2D",
+			"NavigationObstacle2D"
+			"NavigationPolygon",
+			"NavigationRegion2D",
+	} },
+	{ BUILD_OPTION_NAVIGATION_3D, {
+			"NavigationAgent3D",
+			"NavigationLink3D",
+			"NavigationMeshSourceGeometryData3D",
+			"NavigationObstacle3D",
+			"NavigationRegion3D",
+	} },
+	{ BUILD_OPTION_XR, {
+			"XRBodyModifier3D",
+			"XRHandModifier3D",
+			"XRFaceModifier3D",
+			"XRNode3D",
+			"XROrigin3D",
+			"XRInterface",
+			"XRVRS",
+			"XRBodyTracker",
+			"XRControllerTracker",
+			"XRFaceTracker",
+			"XRHandTracker",
+			"XRInterfaceExtension",
+			"XRPose",
+			"XRPositionalTracker",
+			"XRServer",
+			"XRTracker",
+	} },
+	{ BUILD_OPTION_RENDERING_DEVICE, {
+			"RenderingDevice",
+	} },
+	{ BUILD_OPTION_PHYSICS_2D, {
+			"CollisionObject2D",
+			"CollisionPolygon2D",
+			"CollisionShape2D",
+			"Joint2D",
+			"PhysicsServer2D",
+			"PhysicsServer2DManager",
+			"ShapeCast2D",
+			"RayCast2D",
+			"TouchScreenButton",
+	} },
+	{ BUILD_OPTION_PHYSICS_3D, {
+			"CollisionObject3D",
+			"CollisionPolygon3D",
+			"CollisionShape3D",
+			"CSGShape3D",
+			"GPUParticlesAttractor3D",
+			"GPUParticlesCollision3D",
+			"Joint3D",
+			"PhysicalBoneSimulator3D",
+			"PhysicsServer3D",
+			"PhysicsServer3DManager",
+			"PhysicsServer3DRenderingServerHandler",
+			"RayCast3D",
+			"SoftBody3D",
+			"SpringArm3D",
+			"SpringBoneCollision3D",
+			"SpringBoneSimulator3D",
+			"VehicleWheel3D",
+	} },
+	{ BUILD_OPTION_TEXT_SERVER_ADVANCED, {
+			"Control",
```

Comment:
Does this mean Control is removed if you disable TextServerAdvanced? That seems wrong.